talisker/requests.py

```diff
@@ -302,3 +310,137 @@ def enable_requests_logging():  # pragma: nocover
     requests_log = logging.getLogger("requests.packages.urllib3")
     requests_log.setLevel(logging.DEBUG)
     requests_log.propagate = True
+
+
+class TaliskerAdapter(HTTPAdapter):
+
+    def __init__(self, backends=None, connect=1.0, read=10.0, max_retries=0,
+                 *args, **kwargs):
+        # set up backends
+        if backends:
```

Comment:
Extract this to a `_make_backend_iter()` or similar? It feels a bit too chunky for an `__init__`